packages/stream_video/lib/src/stream_video.dart

```diff
@@ -173,46 +195,72 @@ class StreamVideo {
   /// Invoked when a call was created by another user with ringing set as True.
   void Function(Call)? onIncomingCall;
 
-  /// Connects the [user] to the Stream Video service using a [TokenProvider].
-  /// This method is great if your service requires tokens to be refreshed.
-  ///
-  /// For applications outside of test and development, this method may be more
-  /// suitable as token refreshes are handled for you.
-  Future<Result<String>> connectUserWithProvider(
-    UserInfo user, {
-    required TokenProvider tokenProvider,
-  }) async {
-    _logger.i(() => '[connectUser] user.id : ${user.id}');
-    if (currentUser != null) {
-      _logger.w(() => '[connectUser] rejected (already set): $currentUser');
-      final res = await _tokenManager.getToken();
-      return res.map((data) => data.rawValue);
+  /// Connects the user to the Stream Video service.
+  Future<Result<None>> connect() async {
+    final user = _state.currentUser.valueOrNull;
+    if (user == null) {
+      _logger.w(() => '[connect] rejected (currentUser is null)');
+      return Result.error('User is not set');
     }
-    final tokenResult = await _tokenManager.setTokenProvider(
-      user.id,
-      tokenProvider: tokenProvider,
-    );
-    if (tokenResult.isFailure) {
-      return tokenResult.map((data) => data.rawValue);
-    }
-    _state.currentUser.value = user;
+    _logger.i(() => '[connect] currentUser.id : ${user.id}');
 
     try {
-      final result = await _client.connectUser(user);
-      _logger.v(() => '[connectUser] completed: $result');
+      final result = await _client.connectUser(user.info);
+      _logger.v(() => '[connect] completed: $result');
       if (result is Failure) {
         return result;
       }
       _subscriptions.add(_idEvents, _client.events.listen(_onEvent));
       _subscriptions.add(_idAppState, lifecycle.appState.listen(_onAppState));
-      await pushNotificationManager?.onUserLoggedIn();
-      return tokenResult.map((data) => data.rawValue);
+      try {
+        await pushNotificationManager?.onUserLoggedIn();
+      } catch (e, stk) {
+        _logger.w(() => '[connect] #pnManager; failed: $e, $stk');
+      }
+      try {
+        _logger.v(() => '[connect] checking token');
+        await _tokenManager.getToken();
+      } catch (e, stk) {
+        _logger.w(() => '[connect] #tokenManager; failed: $e, $stk');
+      }
```

Comment:
any reason we are fetching token here? The `connectUser` call will do it automatically right?

Reply:
We just want to wait if it's not fetched yet to let `connect` complete when token is available

Reply:
GetToken does not trigger extra requests.
It either triggers request if token is null or waits the previous request to complete.
If tokens exists it will return immediately